libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -472,10 +486,15 @@ public async Task<JToken> ResolveRefAsync(JToken refToken, SourceContext sourceC
                 }
             }
 
+            if (!string.IsNullOrEmpty(resource.FullName) && range.Path == resource.FullName)
+            {
+                _resourceTokenCache.AddOrUpdate(resource.FullName, (json, range), (key, oldValue) => (json, range));
```

Comment:
If we do see that the benchmarks show that there is significant improvement from adding this cache, then we need to test change notification (or think how we'll support users who depend on it). If we have a cache and there is a change in file system, will that change end up in a cache update?

Reply:
Change notifications are wired to each ResourceProvider, and the cache doesn't keep the ResourceProviders from providing change notification for resources they are managing, so any change notifications should still be triggered if a resource were to be modified.